### PR TITLE
docs: Include section on code generation in contributing guide

### LINKED ISSUE #24: how to understand how duckplyr was built?
Here is what I (think I) understand and don't understand. There'd be value in writing it up in a contributing guide for fixing current problems, onboarding collaborators, maybe even inspiring people with similar goals (as a side effect, not a priority).

- code from `tools/` was sourced in alphabetical order.
- what are the subfolders in `tools/`?
- there's something ensuring duckplyr can be synced? at least I see things related to git diff/patch?

### PR DESCRIPTION
**Pull request by krlmlr**
Closes #24.

CC @maelle.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -82,6 +82,13 @@ Let's discuss first!
 
 Let's discuss first!
 
+## Code generation and synchronization
+
+The duckplyr package is a "long-running fork of dplyr", and has code to generate parts of its own implementation and tests, and to synchronize changes from duckplyr with our codebase.
+The main synchronization script is `tools/99-sync.R`, it also contains instructions how to set up a new clone.
+This script should be run whenever the implementation of the verbs changes in a substantial way.
+Watch out for the `# Generated by ...` headers at the top of source files.
```
Comment (maelle) [inline .github/CONTRIBUTING.md:90]: should they use the header that ensure there's a read-only warning in RStudio IDE (and presumedly Positron)?

**Review by maelle — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (krlmlr) [reply .github/CONTRIBUTING.md:90]: What do you mean by "use the header"?

Comment (maelle) [reply .github/CONTRIBUTING.md:90]: if you open NAMESPACE for instance, `# Generated by roxygen2: do not edit by hand` makes a "Read-only" alert appear.

Comment (krlmlr) [reply .github/CONTRIBUTING.md:90]: I guess I see what you mean, we can reword in the next iteration.

**Pull request merged**